src/environment.ts

```diff
@@ -144,114 +76,52 @@ export class SecureEnvironment implements MembraneBroker {
-                        currentGetter = function(this: any): RedValue {
-                            const value: BlueValue = apply(originalOrDistortedGetter, env.getBlueValue(this), emptyArray);
-                            return env.getRedValue(value);
-                        };
-                    }
-
-                    blueDescriptor.get = function(): RedValue {
-                        return apply(currentGetter, this, emptyArray);
+    remap(redValue: RedValue, blueValue: BlueValue, blueDescriptors: PropertyDescriptorMap) {
+        const broker = this;
+        for (const key in blueDescriptors) {
+            if (!canRedPropertyBeTamed(redValue, key)) {
+                console.warn(`Property ${key} of ${redValue} cannot be remapped.`);
+                continue;
+            }
+            // avoid poisoning by only installing own properties from blueDescriptors
+            const blueDescriptor = assign(ObjectCreate(null), blueDescriptors[key]);
+            const redDescriptor = assign(ObjectCreate(null), blueDescriptor);
+            if ('value' in blueDescriptor) {
+                redDescriptor.value = broker.getRedValue(blueDescriptor.value);
+            } else {
+                // Use the original getter to return a red object, but if the 
+                // sandbox attempts to set it to a new value, this mutation will
+                // only affect the sandbox's global object, and the getter will
+                // start returning the new provided value rather than calling onto
+                // the blue realm. This is to preserve the object graph of the
+                // blue realm.
+                let currentBlueGetter: (this: RedValue) => RedValue = () => undefined;
+
+                if (isFunction(blueDescriptor.get)) {
+                    const { get: blueGetter } = blueDescriptor;
+                    const blueDistortedGetter: () => BlueValue = WeakMapGet(this.distortionMap, blueGetter) || blueGetter;
+                    currentBlueGetter = function() {
+                        const value: BlueValue = apply(blueDistortedGetter, broker.getBlueValue(this), emptyArray);
+                        return broker.getRedValue(value);
+                    };
+                    redDescriptor.get = function(): RedValue {
+                        return apply(currentBlueGetter, this, emptyArray);
                     };
-
-                    if (isFunction(blueDescriptor.set)) {
-                        blueDescriptor.set = function(v: RedValue): void {
-                            // if a global setter is invoke, the value will be use as it is as the result of the getter operation
-                            currentGetter = () => v;
-                        };
-                    }
                 }
 
-                const redDescriptor = ReflectGetOwnPropertyDescriptor(redValue, key);
-                if (!isUndefined(redDescriptor) && 
```

Comment:
all these debug-mode messages are now removed... instead, we just want when the descriptor is non-configurable